webapp/app.py

```diff
@@ -198,7 +198,16 @@ def bad_request_error(error):
 
 @app.errorhandler(SecurityAPIError)
 def security_api_error(error):
-    return flask.render_template("security-error.html"), 500
+
+    message = error.response.json().get("message")
+
+    if error.response.status_code == 404:
+        return flask.render_template("404.html", message=message), 404
```

Comment:
I did write a message about "why not use `flask.abort`?", then I realised it's because that doesn't work when you're already within an error handler. This is a good solution.

I'm wondering, though, if this highlights that we're doing the wrong thing a bit? If what we really want to say is that something is just missing, maybe raising is as a `SecurityAPIError` is fundamentally misleading.

I think I'm going to merge this so we can get it out and then file a follow-up PR to try a restructure and then you can review it and see what you think. Sorry for all the back and forth.